deeptime/markov/msm/tram/_tram.py

```diff
@@ -0,0 +1,708 @@
+            chosen lagtime.
+        """
+        # count all transitions and state counts, without restricting to connected sets
+        self._largest_connected_set = self._find_largest_connected_set(ttrajs, dtrajs, bias_matrices)
+
+        # get rid of any state indices that do not belong to the largest connected set
+        dtrajs = self._restrict_to_connected_set(dtrajs)
+
+        # In the case of replica exchange: get all trajectory fragments (also removing any negative state indices).
+        # the dtraj_fragments[k] contain all trajectories within thermodynamic state k, starting a new trajectory at
+        # each replica exchange swap. If there was no replica exchange, dtraj_fragments == dtrajs.
+        dtraj_fragments = self._get_trajectory_fragments(dtrajs, ttrajs)
+
+        # ... convert those into count matrices.
+        state_counts, transition_counts = self._make_count_models(dtraj_fragments)
+
+        return dtrajs, state_counts, transition_counts
+
+    def _find_largest_connected_set(self, ttrajs, dtrajs, bias_matrices):
+        """ Find the largest connected set of Markov states based on the connectivity settings and the input data.
+        A full counts model is first calculated from all dtrajs. Then, the connected set is computed based on
+        self.connectivity and self.connectivity_factor. The full counts model is then reduces to the connected set and
+        returned. If the connectivity setting is unknown, the full counts model is returned.
+
+        Parameters
+         ----------
+         dtrajs: ndarray
+            The discrete trajectories in the form an 2-d integer ndarray. dtrajs[i] contains one trajectory.
+            dtrajs[i][n] contains the Markov state index that the n-th sample from the i-th trajectory was binned
+            into. Each of the dtrajs can be of variable length.
+         bias_matrices: ndarray
+            The bias energy matrices. bias_matrices[i, n, l] contains the bias energy of the n-th sample from the i-th
+            trajectory, evaluated at thermodynamic state k. The bias energy matrices should have the same size as
+            dtrajs in both the 0-th and 1-st dimension. The seconds dimension of of size n_therm_state, i.e. for each
+            sample, the bias energy in every thermodynamic state is calculated and stored in the bias_matrices.
+         ttrajs: ndarray, optional
+            ttrajs[i] indicates for each sample in the i-th trajectory what thermodynamic state that sample was
+            sampled at. If ttrajs is None, we assume no replica exchange was done. In this case we assume each
+            trajectory  corresponds to a unique thermodynamic state, and n_therm_states equals the size of dtrajs.
+
+        Returns
+        -------
+        counts_model: MarkovStateModel
+            The counts model pertaining to the largest connected set.
+        """
+        estimator = TransitionCountEstimator(lagtime=self.lagtime, count_mode=self.count_mode)
```

Comment:
what would one have to do to support different count modes?

Reply:
nothing, they are supported

Reply:
Aren't they excluded in the constructor?